src/app/pages/storage/modules/disks/components/smart-test-result-list/smart-test-result-list.component.ts

```diff
@@ -0,0 +1,157 @@
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SmartTestResultListComponent implements OnInit {
+  @Input() type: SmartTestResultPageType;
+  @Input() pk: string;
+  queryParams: QueryParams<SmartTestResults> = [];
+  disks: Disk[] = [];
+  smartTestResults: SmartTestResultsRow[];
+  filterString = '';
+  dataProvider: AsyncDataProvider<SmartTestResultsRow>;
+  protected readonly requiredRoles = [Role.FullAdmin];
+
+  columns = createTable<SmartTestResultsRow>([
+    textColumn({
+      title: this.translate.instant('Disk'),
+      propertyName: 'disk',
+      sortable: true,
+    }),
+    textColumn({
+      title: this.translate.instant('Description'),
+      propertyName: 'description',
+      sortable: true,
+    }),
+    stateButtonColumn({
+      title: this.translate.instant('Status'),
+      propertyName: 'status',
+      sortable: true,
+    }),
+    textColumn({
+      title: this.translate.instant('Remaining'),
+      propertyName: 'remaining',
+      sortable: true,
+      getValue: (row) => {
+        return row.remaining || row.status_verbose;
+      },
+    }),
+    textColumn({
+      title: this.translate.instant('Lifetime'),
+      propertyName: 'lifetime',
+      sortable: true,
+      getValue: (row) => {
+        return row.lifetime || this.translate.instant('N/A');
+      },
+    }),
+    textColumn({
+      title: this.translate.instant('Error'),
+      propertyName: 'lba_of_first_error',
+      sortable: true,
+      getValue: (row) => {
+        return row.lba_of_first_error || this.translate.instant('No errors');
+      },
+    }),
+  ], {
+    rowTestId: (row) => `smart-test-result-${row.disk}-${row.num}`,
+  });
+
+  get hiddenColumns(): Column<SmartTestResultsRow, ColumnComponent<SmartTestResultsRow>>[] {
+    return this.columns.filter((column) => column?.hidden);
+  }
+
+  get diskNames(): string[] {
+    return this.disks.filter((disk) => disk.pool === this.pk).map((disk) => disk.name);
+  }
+
+  constructor(
+    private cdr: ChangeDetectorRef,
+    private ws: WebSocketService,
+    private translate: TranslateService,
+    protected emptyService: EmptyService,
+  ) {}
+
+  ngOnInit(): void {
+    this.createDataProvider();
+  }
+
+  createDataProvider(): void {
+    const smartTestResults$ = this.ws.call('disk.query', [[], { extra: { pools: true } }]).pipe(
+      switchMap((disks) => {
+        this.disks = disks;
+        this.updateQueryParams();
+        return this.ws.call('smart.test.results', this.queryParams);
+      }),
+      map((smartTestResults: SmartTestResults[]) => {
+        const rows: SmartTestResultsRow[] = [];
+        smartTestResults.forEach((smartTestResult) => {
+          smartTestResult?.tests.forEach((test) => {
+            rows.push({ ...test, disk: smartTestResult.disk });
+          });
+        });
+        return rows;
+      }),
+      tap((smartTestResults) => this.smartTestResults = smartTestResults),
```

Comment:
I don't suppose we can rely on `setFilter` in the data provider?

Reply:
Never seen this method before. I'll note this to [Standardize common patterns in ix-table2](https://ixsystems.atlassian.net/browse/NAS-127168)